ctf/views.py

```diff
@@ -59,16 +66,18 @@ def flag(request, ctf_slug, flag_slug):
                     logger.info(f'FLAG: {flag.title} USER: {request.user} INPUT: {flag_input}')
                     flag = Flag.objects.filter(ctf=ctf, slug=flag_slug, flag=flag_input)
                     if flag.exists():
+                        request.session['flag_valid'] = True
                         if user_solved or flag.first().solver:
                             # User has already solved a flag or flag is already solved
-                            return form_valid(request, context)
+                            return redirect(request.path)
                         flag.update(solver=request.user, solved_date=datetime.datetime.now())
                         logger.info(f'Solver: {flag.first().solver}')
                         context['flag'] = flag.first()
                         context['solved'] = True
```

Comment:
Is this needed anymore?

Reply:
Seems to be tied to the logic of how the page is rendered. It doesn't seem to affect any functionality since the page is reloaded anyways.